myhdl/_ShadowSignal.py

```diff
@@ -244,13 +244,15 @@ def _resolve(self):
     def toVerilog(self):
         lines = []
         for d in self._drivers:
-            lines.append("assign %s = %s;" % (self._name, d._name))
+            if d._driven:
+                lines.append("assign %s = %s;" % (self._name, d._name))
```

Comment:
If a driver is declared with `dSig = sig.driver();`, but no logic writes to `dSig`, `dSig._driven` would be `False`. It is not necessary to generate the verilog code `assign sig = dSig;` in this case. That is, if `_driven` is reliable.

My testcase does not cover it, but I've tried this locally by adding the `Y_dummy` line to `test_tristate.py`:
```python
    Y_d = Y.driver()
    Y_dummy = Y.driver()
```